modules/navigation_2d/2d/godot_navigation_server_2d.cpp

```diff
@@ -0,0 +1,1394 @@
+}
+
+COMMAND_2(obstacle_set_radius, RID, p_obstacle, real_t, p_radius) {
+	ERR_FAIL_COND_MSG(p_radius < 0.0, "Radius must be positive.");
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL(obstacle);
+
+	obstacle->set_radius(p_radius);
+}
+
+real_t GodotNavigationServer2D::obstacle_get_radius(RID p_obstacle) const {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, 0);
+
+	return obstacle->get_radius();
+}
+
+COMMAND_2(obstacle_set_velocity, RID, p_obstacle, Vector2, p_velocity) {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL(obstacle);
+
+	obstacle->set_velocity(p_velocity);
+}
+
+Vector2 GodotNavigationServer2D::obstacle_get_velocity(RID p_obstacle) const {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, Vector2());
+
+	return obstacle->get_velocity();
+}
+
+COMMAND_2(obstacle_set_position, RID, p_obstacle, Vector2, p_position) {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL(obstacle);
+	obstacle->set_position(p_position);
+}
+
+Vector2 GodotNavigationServer2D::obstacle_get_position(RID p_obstacle) const {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, Vector2());
+
+	return obstacle->get_position();
+}
+
+COMMAND_2(obstacle_set_avoidance_layers, RID, p_obstacle, uint32_t, p_layers) {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL(obstacle);
+	obstacle->set_avoidance_layers(p_layers);
+}
+
+uint32_t GodotNavigationServer2D::obstacle_get_avoidance_layers(RID p_obstacle) const {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, 0);
+
+	return obstacle->get_avoidance_layers();
+}
+
+void GodotNavigationServer2D::obstacle_set_vertices(RID p_obstacle, const Vector<Vector2> &p_vertices) {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL(obstacle);
+	obstacle->set_vertices(p_vertices);
+}
+
+Vector<Vector2> GodotNavigationServer2D::obstacle_get_vertices(RID p_obstacle) const {
+	NavObstacle2D *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, Vector<Vector2>());
+
+	return obstacle->get_vertices();
+}
+
+void GodotNavigationServer2D::flush_queries() {
+	// In c++ we can't be sure that this is performed in the main thread
+	// even with mutable functions.
+	MutexLock lock(commands_mutex);
+	MutexLock lock2(operations_mutex);
+
+	for (SetCommand2D *command : commands) {
+		command->exec(this);
+		memdelete(command);
+	}
+	commands.clear();
+}
+
+COMMAND_1(free, RID, p_object) {
+	if (geometry_parser_owner.owns(p_object)) {
+		RWLockWrite write_lock(geometry_parser_rwlock);
+
+		NavMeshGeometryParser2D *parser = geometry_parser_owner.get_or_null(p_object);
+		ERR_FAIL_NULL(parser);
+
+		generator_parsers.erase(parser);
+#ifndef CLIPPER2_ENABLED
```

Comment:
There seems to be a typo here. It seems like it should be `#ifdef` instead of `#ifndef`.